packages/container/lib/container.js

```diff
@@ -421,15 +421,16 @@ function deprecatedFactoryFor(container, fullName, options = {}) {
   }
 
   let cache = container.factoryCache;
-  if (cache[fullName]) {
-    return cache[fullName];
+  let cached = cache[fullName];
+  if (cached) {
+    return cached;
   }
   let factory = registry.resolve(fullName);
   if (factory === undefined) {
     return;
   }
 
-  let type = fullName.split(':')[0];
+  let [type] = fullName.split(':');
```

Comment:
probably it is created once for such cases and reused, https://cl.ly/2O0P2r0a3Z0Q, and such usage of spread exist in other places so that silly helper probably already exist :P